Add end_month and end_day feature to setup_fiscal_calendar

qwertytam opened the pull request:
Following on from discussion on #20 , adding support for `end_month` and `end_day` feature to `setup_fiscal_calendar`. My coding skillz are far from brilliant, so welcome any/all suggestions for improvement.

nicmendoza commented on fiscalyear.py:
```diff
@@ -53,8 +100,12 @@ def _validate_fiscal_calendar_params(start_year, start_month, start_day):
+    """Modify the fiscal calendar.
+
+    If either of ``start_month`` or ``start_day`` are present, ``end_month``
+    and ``end_day`` are ignored.
```
This might be surprising behavior. I think it would be preferable to throw an error when invalid combinations of arguments are supplied

qwertytam commented:
Added logic to check for invalid combination of supplied arguments with #6955029c4ad1d0eb42c7c7a631281ec3f4459bd1

Implemented logic in new function `_check_only_start_or_end_present`. My idea is once this pull request is approved, can then copy the same logic flow to the `fiscal_calendar` decorator.
